File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        int start = 0;
        int end = nums.length-1;

        while(start<=end)
        {
            if((nums[start]+nums[end])==target)
            {
                if(nums[start]<nums[end])
                {
                    return  new int[]{start+1,end+1};
                }
                else{
                    return   new  int[]{end +1, start+1};
                }
               
            }
            if((nums[start]+nums[end])>target)
            {
                end--;
            }
            else{
                start++;
            }
        }

        return new int[]{-1,-1};
    }
}

File: Data Structures & Algorithms/subsets/submission-0.py
class Solution:
    def subsets(self, nums: List[int]) -> List[List[int]]:
        res=[]
        n=len(nums)

        def backtrack(index,path):
            # reach end 
            if index==n:
                res.append(path[:])
                return

            # include
            path.append(nums[index])
            backtrack(index+1,path)

            # exclude
            path.pop()
            backtrack(index+1,path)

        backtrack(0,[])
        return res
        